### PR TITLE
More docker resiliency for tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
Probably fixes https://github.com/aspnet/AspNetCore-Internal/issues/2502

### CODE REVIEW

File: src/SignalR/server/StackExchangeRedis/test/Docker.cs [hunk 1]
```diff
@@ -81,6 +81,9 @@ private void StartRedis(ILogger logger)
             {
                 logger.LogError(ex, "Error starting redis docker container, retrying.");
                 Thread.Sleep(1000);
+
+                // Call stop just in case the container somehow started after the timeout so our retry logic doesn't fail
+                RunProcessAndWait(_path, $"stop {_dockerContainerName}", "docker stop", logger, TimeSpan.FromSeconds(15), out var _);
```
Comment (anurse) [inline src/SignalR/server/StackExchangeRedis/test/Docker.cs]: Will this throw if the container is already stopped?

Comment (BrennanConroy) [reply src/SignalR/server/StackExchangeRedis/test/Docker.cs]: Nope, that would be `RunProcessAndThrowIfFailed`

**Comment by BrennanConroy** [comment]
Ping

**Review by analogrelay — APPROVED** [review]

**Comment by BrennanConroy** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=195874&view=logs

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2490

**Pull request merged**